### PR TITLE
Updating the tie-breaking for MD to match Picard.

### PR DESCRIPTION
**Pull request by davidadamsphd**

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSparkUtils.java [hunk 1]
```diff
@@ -100,8 +101,13 @@
                 out.add(pair.first());
             }
 
-            // order by score
-            List<PairedEnds> scored = Ordering.natural().reverse().onResultOf((PairedEnds pair) -> pair.score()).sortedCopy(paired.get(true));
+            // Order by score using ReadCoordinateComparator for tie-breaking.
+            Comparator<PairedEnds> pairedEndsComparator =
+                    Comparator.comparing((java.util.function.Function<PairedEnds, Integer>) (pairedEnds1) -> -pairedEnds1.score()).thenComparing((o1, o2) -> {
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSparkUtils.java]: I think it reads better as
```
          Comparator<PairedEnds> pairedEndsComparator =
                    Comparator.comparing(PairedEnds::score).reversed()
                        .thenComparing((o1, o2) -> new ReadCoordinateComparator(header).compare(o1.first(), o2.first()));
```

but that might just be be nitpicking

**Comment by lbergelson** [comment]
Could you add a test that demonstrates the correct sorting order?

File: src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSparkUtils.java (hunk 1 above)
Comment (davidadamsphd) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSparkUtils.java]: Fixed.

**Comment by davidadamsphd** [comment]
I'm going to punt on the test because I'll be creating a large PR just for unit tests for MD later this week.

**Comment by lbergelson** [comment]
fine, go ahead

**Pull request merged**